core/src/main/scala/org/apache/spark/h2o/converters/SparkDataFrameConverter.scala

```diff
@@ -52,7 +52,7 @@ private[h2o] object SparkDataFrameConverter extends Logging {
   def toH2OFrame(hc: H2OContext, dataFrame: DataFrame, frameKeyName: Option[String]): H2OFrame = {
     import H2OSchemaUtils._
     // Flatten the Spark data frame so we don't have any nested rows
-    val flatDataFrame = flattenDataFrame(dataFrame)
+    val flatDataFrame = flattenStructsInDataFrame(dataFrame)
     val dfRdd = flatDataFrame.rdd
     val keyName = frameKeyName.getOrElse("frame_rdd_" + dfRdd.id + Key.rand())
 
```

Comment:
If I'm correct, the flatting method you introduced also handles arrays of different size. If if would be also handle vectors as described in the last comment, we could actually simplify the code starting from line 59 a lot ( maybe actually remove). In order to do so, the output of the flatten method should be:

Maps, Structs, Arrays, Dense vectors should be flatten. Sparse vectors should not be flatten, but should be also at the top-level schema structure and should have the same max size.

Reply:
Yes, you're 100% correct. Let's do such changes once the general flattening is merged. :-)

Reply:
Cool!